src/model/constraints/behavior_constraints.py

```diff
@@ -0,0 +1,29 @@
+from src.model.constraints import utils
+from src.model.constraints.constraint import Constraint
+
+class Behavior_Con(Constraint):
+    pass
+    
+class Danger_Con(Behavior_Con):
+    pass
+
+class Does_Maneuver_Con(Behavior_Con):
+
+    def __init__(self, parent, actors, maneuver):
+        self.arity = 2
+        # self.type_id = ? # TODO
+        self.maneuver = maneuver # TODO: Confirm
+        super().__init__(parent, "Does_Maneuver", actors, None) # TODO: Confirm
+
+    def get_heuristic_value(self):
+        return 0 # TODO
+
+class Collision_Con(Danger_Con):
+
+    def __init__(self, parent, actors):
+        self.arity = 2
+        # self.type_id = ? # TODO
+        super().__init__(parent, "Collision", actors, None) # TODO: Confirm
+
+    def get_heuristic_value(self):
+        return 0 # TODO
```

Comment:
In search/mhs/utils.py, getHeuristic() (line 169) expects a get_heuristic_value() for all constraints. Do we have heuristic values for behavior constraints? If so, how do we evaluate them?

Reply:
The get_heuristic_value is only used in the MHS mode of the solver. Currently, the behavior-related things will only be allowed for the other mode, which is "brute" or "complete". I think the best thing to o for now is to throw some error if the get_heuristic_value is called on invalid constraint types.

Reply:
But then things won't run through when we have behavior constraints for now

Reply:
As we discussed last time, I separated static and behaviour constraints and added different abstract methods—see the latest commit https://github.com/ArenBabikian/concretize/pull/1/commits/5e41adcd6acfd4dfd318bf989c5e4550b6f84863